crates/burn-core/src/module/param/base.rs

```diff
@@ -33,10 +32,10 @@ use core::ops::Deref;
 pub struct Param<T: Parameter> {
     /// The unique ID of this parameter. This is used by eg. optimizers to associate a gradient with a specific parameter.
     pub id: ParamId,
-    state: OnceCell<T>,
+    state: SyncOnceCell<T>,
```

Comment:
The goal of using `OnceCell` is that we don't have to lock each time we access a parameter, only during lazy initialization. By definition, `Param` can't be `Sync`, and it's intended that way because almost all tensor operations are owned and consume the tensor. When you clone a parameter, we actually initialize it so that both versions have the same state. If you could send it to another thread without cloning it (i.e., implementing `Sync`), then the initialization could be triggered on both threads at the same time, and we would need a synchronization method.